nav2_collision_monitor/include/nav2_collision_monitor/types.hpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2022 Samsung Research Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_COLLISION_MONITOR__TYPES_HPP_
+#define NAV2_COLLISION_MONITOR__TYPES_HPP_
+
+#include <vector>
+#include <mutex>
+#include <cmath>
+
+#include "rclcpp/rclcpp.hpp"
+
+namespace nav2_collision_monitor
+{
+
+// Mutex type is variable for future code flexibility
+typedef std::recursive_mutex mutex_t;
+
+// Velocity for 2D model of motion
+struct Velocity
+{
+  double x;  // x-component of linear velocity
+  double y;  // y-component of linear velocity
+  double tw;  // z-component of angular twist
+
+  inline bool operator!=(const Velocity & second) const
+  {
+      return x != second.x || y != second.y || tw != second.tw;
+  }
+
+  inline bool operator<(const Velocity & second) const
+  {
+    const double first_vel = x*x + y*y;
+    const double second_vel = second.x*second.x + second.y*second.y;
+    return first_vel < second_vel;
+  }
+
+  inline Velocity& operator+=(const double & inc)
```

Comment:
When is this used? This seems inexact and should be removed

Reply:
It is used in `APPROACH` model when gradually releasing robot from its slowdown to normal operation:
```
      // Release robot from approach: gradually increase robot speed in order to avoid twitching
      const Velocity release_vel = ra_prev_.req_vel+=release_step_;
      if (release_vel < max_vel) {
        ra.req_vel = release_vel;
      } else {
        ra.req_vel = max_vel;
        // Robot already reached its normal speed: releasing its to usual operation
        release_operation_ = false;
      }
```
This gradual speed increase is some kind of smoothing, made to wipe out the twitching of the robot conditioned by sensors data noise.

Reply:
OK but overriding the operator with this isn't great since this is not an exact expression of `+=` of those terms, its more heuristic-y that it works but there's not a way to  consistently and properly add some quantity to XY fields as `m/s` and then $\theta$ as `rad/s` units

Reply:
Okay, all `+=` logic was moved to a new `releaseOperation()` method: https://github.com/ros-planning/navigation2/pull/2982/files#diff-e1c76d88179c14a1cf399e94bebd4f5d17dedca39e09b3bbdc810aa11a705543R462